src/index.ts

```diff
@@ -0,0 +1,25 @@
+import { createClient } from "@supabase/supabase-js";
+import { createPlugin } from "@ubiquity-os/ubiquity-os-kernel";
+import { ExecutionContext } from "hono";
+import manifest from "../manifest.json";
+import { createAdapters } from "./adapters";
+import { run } from "./run";
+import { SupportedEvents } from "./types/context";
+import { Database } from "./types/database";
+import { Env } from "./types/env";
+import { PluginSettings } from "./types/plugin-input";
+
+const pluginPromise = createPlugin<PluginSettings, Env, SupportedEvents>(
+  (context) => {
+    const supabase = createClient<Database>(context.env.SUPABASE_URL, context.env.SUPABASE_KEY);
+    return run({ ...context, adapters: createAdapters(supabase, context) });
+  },
+  // @ts-expect-error strings cannot be assigned to events
+  manifest
+);
+
+export default {
+  async fetch(request: Request, env: Env, context: ExecutionContext) {
+    return (await pluginPromise).fetch(request, env, context);
+  },
+};
```

Comment:
I think you can just export the result from `createPlugin` because it returns a Hono app instance and you can just [default export](https://hono.dev/docs/getting-started/cloudflare-workers#_2-hello-world)

Reply:
I tried that way at first and on deploy I got https://github.com/Meniole/command-query-user/actions/runs/11323127167/job/31485152893#step:6:75

Will attemp again, might have messed up somewhere.